### ISSUE TITLE
Scope docstring is confusing/wrong?

### ISSUE THREAD

**Issue by lidavidm** [open-telemetry/opentelemetry-cpp#2508]
**Is your feature request related to a problem?**

Scope is documented as "On destruction, the given span is ended":

https://github.com/open-telemetry/opentelemetry-cpp/blob/8da71b78ed01bd1e40861b0ab50c1362e03abf15/api/include/opentelemetry/trace/scope.h#L18-L24

However, from looking through the implementation, it _appears_ only to reset the active span and doesn't actually end the span? Examples in documentation align with this (e.g. creating a scope, then explicitly ending the span). Also, if it _did_ end the span, that would mean there is no (obvious) way to activate a span without ending it.

**Describe the solution you'd like**
Update the docstring.

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

**Comment by lalitb** [comment]
Agree, feel free to create a PR with the fix. Thanks for reporting.
